lambda/save-raw-events/tests/save-raw-events.test.ts

```diff
@@ -0,0 +1,229 @@
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
+        services: [
+          {
+            client_id: "client_id",
+            timestamp: new Date().toISOString,
+            component_id: "component_id",
+            user: {
+              user_id: "user_id",
+            },
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrowError();
+  });
+  test("throws error when component_id is missing", () => {
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
+        services: [
+          {
+            client_id: "client_id",
+            timestamp: new Date().toISOString,
+            event_name: "event_name",
+            user: {
+              user_id: "user_id",
+            },
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrowError();
+  });
+  test(" throws error when user is missing", () => {
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
+        services: [
+          {
+            client_id: "client_id",
+            timestamp: new Date().toISOString,
+            event_name: "event_name",
+            component_id: "component_id",
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrowError();
+  });
+  test("throws error when user_id  is missing", () => {
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
+        services: [
+          {
+            client_id: "client_id",
+            timestamp: new Date().toISOString,
+            event_name: "event_name",
+            component_id: "component_id",
+            user: {},
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrowError();
+  });
+});
+
+describe("handler", () => {
+  beforeEach(() => {
+    dynamoMock.reset();
+    sqsMock.reset();
+    process.env.TABLE_NAME = "TABLE_NAME";
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("Adds raw event to the table", async () => {
+    await handler(TEST_SQS_EVENT);
+    expect(dynamoMock.commandCalls(PutCommand).length).toEqual(1);
+  });
+});
+
+describe("handler error handling", () => {
+  let consoleErrorMock: jest.SpyInstance;
+
+  beforeEach(() => {
+    dynamoMock.reset();
+    sqsMock.reset();
+    process.env.TABLE_NAME = "TABLE_NAME";
+    consoleErrorMock = jest.spyOn(global.console, "error").mockImplementation();
+    dynamoMock.rejectsOnce("mock error");
+  });
+  afterEach(() => {
+    consoleErrorMock.mockRestore();
+    jest.clearAllMocks();
+  });
+  test("logs the error message", async () => {
+    await handler(TEST_SQS_EVENT);
+    expect(consoleErrorMock).toHaveBeenCalledTimes(1);
+  });
+  test("sends the event to the dead letter queue", async () => {
+    await handler(TEST_SQS_EVENT);
+    expect(sqsMock.commandCalls(SendMessageCommand).length).toEqual(1);
```

Comment:
Take a look and see if Alex's suggestion here makes sense for this test too?
https://github.com/alphagov/di-account-management-backend/pull/38#discussion_r1019990660

I thought the adjustment ended up with a better test in the end.
You can see my change here:
https://github.com/alphagov/di-account-management-backend/commit/24bd4947089d12ba04b9c3e2e99f662236e3f3ea
`lambda/delete-user-services/tests/delete-user-services.test.ts`